qt/icons/src/CharIconPainter.cpp

```diff
@@ -0,0 +1,74 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidQtIcons/CharIconPainter.h"
+#include "MantidQtIcons/Icon.h"
+#include <QVariant>
+#include <cmath>
+
+namespace MantidQt {
+namespace Icons {
+
+void CharIconPainter::paint(IconicFont *iconic, QPainter *painter, QRect rect,
+                            QIcon::Mode mode, QIcon::State state,
+                            QList<QHash<QString, QVariant>> &options) {
+  for (auto &option : options) {
+    this->paintIcon(iconic, painter, rect, mode, state, option);
+  }
+}
+
+void CharIconPainter::paintIcon(IconicFont *iconic, QPainter *painter,
+                                QRect rect, QIcon::Mode mode,
+                                QIcon::State state,
+                                QHash<QString, QVariant> &options) {
+  // These variables will be useful later on, so I have left them being passed
+  // down this far. This is because they can be used for defining variable
+  // changes based on state of the buttons/QObject that the Icon is present in.
+  // Since we currently don't use this feature availible with QIcon they have
+  // not yet been implemented.
+  UNUSED_ARG(mode);
+  UNUSED_ARG(state);
+
+  painter->save();
+  const auto colorVariant = options[QString("color")];
+  const auto scaleVariant = options[QString("scaleFactor")];
```

Comment:
Thinking about it, can we make a check for `options` containing the keys, and throw something if it doesn't?

Reply:
The values are defaulted higher up the chain so they will always be present. Since these are hard coded values I don't see much point.